[WPE][GTK][JHBuild] Update gstreamer and related libraries and add sparkle-cdm

clopez opened the pull request:
#### ed54b0254b83b2ea15aa23354e1e0c91f1d8dfaf
<pre>
[WPE][GTK][JHBuild] Update gstreamer and related libraries and add sparkle-cdm
<a href="https://bugs.webkit.org/show_bug.cgi?id=266229">https://bugs.webkit.org/show_bug.cgi?id=266229</a>

Reviewed by Michael Catanzaro.

This updates GStreamer and related libraries in order to
be able to get the default build with -DENABLE_EXPERIMENTAL_FEATURES=ON
working on Ubuntu 22.04.

This also simplifies the GStreamer fetch/build process by using the mono-repo
instead of the split tarballs.

It also fixes a build error with libaom and ARM32 and adds Sparkle-CDM
in order to build with -DENABLE_THUNDER=ON.

* Tools/gstreamer/jhbuild.modules:
* Tools/gtk/jhbuild.modules:
* Tools/wpe/jhbuild.modules:

Canonical link: <a href="https://commits.webkit.org/271940@main">https://commits.webkit.org/271940@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b0f98fea045199694bc30a4f1b123122b4040abb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/30128 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/8793 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/31442 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/32632 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/27243 "Built successfully") 
| [❌ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/30803 "Failed to checkout and rebase branch from PR 21639") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/10993 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/51/builds/6038 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/27253 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/11/builds/30428 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/7376 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/25696 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/6305 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/6458 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/26789 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33969 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/27469 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/27222 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/32637 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/6406 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/4574 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/30439 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/8156 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/7160 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3886 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6935 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Tools/gstreamer/jhbuild.modules:
```diff
@@ -156,6 +106,14 @@
+    <branch repo="github-tarball"
+            module="Sparkle-CDM/sparkle-cdm/tarball/${version}"
+            checkoutdir="sparkle-cdm"
+            version="3e1e0dcb221d044213f8c6e32b0b06102e7fb224"
```
I think the version is supposed to be numeric, so I would omit this. What it means is "if this version or a greater version is already installed as a system dependency, then do not download and build." This doesn't work for git hashes.

mcatanzaro replied:
You can just copy the hash into the `module=` line instead

clopez replied:
AFAIK that version check is only performed when the entry has a `<pkg-config>` entry.
I tried removing the version and moving it into the module but then JHBuild fails to parse the xml, seems it expects a version number.
So I will just provide a version number based on the commit date (20231106)

clopez committed:
[WPE][GTK][JHBuild] Update gstreamer and related libraries and add sparkle-cdm
https://bugs.webkit.org/show_bug.cgi?id=266229

Reviewed by Michael Catanzaro.

This updates GStreamer and related libraries in order to
be able to get the default build with -DENABLE_EXPERIMENTAL_FEATURES=ON
working on Ubuntu 22.04.

This also simplifies the GStreamer fetch/build process by using the mono-repo
instead of the split tarballs.

It also fixes a build error with libaom and ARM32 and adds Sparkle-CDM
in order to build with -DENABLE_THUNDER=ON.

* Tools/gstreamer/jhbuild.modules:
* Tools/gtk/jhbuild.modules:
* Tools/wpe/jhbuild.modules:

Canonical link: https://commits.webkit.org/271940@main
